### PR TITLE
Upgrade operator-sdk to 1.17.0

### LINKED ISSUE #56: Upgrade operator-sdk to v1.17.0
Operator-sdk has several improvements in 1.17.0, including support for k8s 1.23 [1].
This upgrade will include several dependency changes and upgrade golang to 1.17.

[1] https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/

/kind cleanup

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

Upgrade to operator-sdk v1.17.0, along with dependencies for controller-runtime and kubernetes:

- Update golang dependencies:
  - k8s.io/* packages to v1.23.0
  - controller-runtime to v0.11.0
  - ginkgo to v1.16.5
  - gomega to v1.17.0
  - manifestival to v0.7.1
- Update tooling used for manifest generation and testing:
  - Upgrade controller-gen to v0.8.0
  - Upgrade opm to v1.19.1
  - Upgrade operator-sdk to v1.17.0
- Upgrade envtest to the latest version that runs as a go binary. Fix test setup to use a cancellable context. Removed our custom script to run envtest.
- Refactored the Makefile so it aligns with what is generated by operator-sdk and controller-runtime.
- Added upstream bug fixes to the operator deployment manifest.

Fixes #56 

/kind cleanup

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/master/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
Upgrade operator-sdk to v1.17.0, controller-runtime to v0.8.0, and k8s.io/* to v1.23.0
```

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=shipwright-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.13.0+git
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.16.0+git
```
Comment (otaviof) [inline bundle.Dockerfile:9]: The PR title says `1.17.0`, but I only see references to `1.16.0` instead.

**Comment by gabemontero** [comment]
LGTM aside from @otaviof 's question at https://github.com/shipwright-io/operator/pull/59#discussion_r826869340

if there is an explanation for that @adambkaplan go ahead and post a comment in that thread;  otherwise, it does seem like it should be consistent and go to 17 instead of 16

File: bundle.Dockerfile (hunk 1 above)
Comment (adambkaplan) [reply bundle.Dockerfile:9]: Yeah, this seems to be a bug in operator-sdk. The version of the label and the sdk version are always off by 1.

**Comment by adambkaplan** [comment]
Filed https://github.com/operator-framework/operator-sdk/issues/5612 to report the semantic version issue.

**Review by adambkaplan — COMMENTED** [review]
/approve

Self-approving

**Pull request merged**